pkg/detectors/npm/npm_token_v1_test.go

```diff
@@ -0,0 +1,202 @@
+		},
+
+		// Invalid
+		"invalid/_authToken/variable": {
+			input: `//npm.pkg.github.com/:_authToken=${GITHUB_PACKAGES_AUTH_TOKEN}`,
+		},
+		"invalid/default": {
+			input: `assert(registry, 'registry not set, example: "https://nexus.foo.com/repository/mynpm/"')
+const tokenErrorMsg =  'npm token invalid, example: "NpmToken.00000000-0000-0000-0000-000000000000" before base64 encoded'`,
+		},
+		"invalid/not_uuid": {
+			input: `# .npmrc
+# @ngiq:registry=https://registry.corp.net/repository/npm-group
+# //registry.corp.net/repository/:_authToken=NpmToken.xxxx`,
+		},
+	}
+	testPattern(t, ScannerV1{}, cases)
+}
+
+func TestNpmTokenV1_FromChunk(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	testSecrets, err := common.GetSecret(ctx, "trufflehog-testing", "detectors2")
+	if err != nil {
+		t.Fatalf("could not get test secrets from GCP: %s", err)
+	}
+	secret := testSecrets.MustGetField("NPM_TOKEN_V1")
+	inactiveSecret := testSecrets.MustGetField("NPM_TOKEN_V1_INACTIVE")
+
+	type args struct {
+		ctx    context.Context
+		data   []byte
+		verify bool
+	}
+	tests := []struct {
+		name    string
+		s       ScannerV1
+		args    args
+		want    []detectors.Result
+		wantErr bool
+	}{
+		{
+			name: "found, verified",
+			s:    ScannerV1{},
+			args: args{
+				ctx:    context.Background(),
+				data:   []byte(fmt.Sprintf("You can find a npm secret %s within", secret)),
+				verify: true,
+			},
+			want: []detectors.Result{
+				{
+					DetectorType: detectorspb.DetectorType_NpmToken,
+					Verified:     true,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "found, unverified",
+			s:    ScannerV1{},
+			args: args{
+				ctx:    context.Background(),
+				data:   []byte(fmt.Sprintf("You can find a npm secret %s within but not valid", inactiveSecret)), // the secret would satisfy the regex but not pass validation
+				verify: true,
+			},
+			want: []detectors.Result{
+				{
+					DetectorType: detectorspb.DetectorType_NpmToken,
+					Verified:     false,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "not found",
+			s:    ScannerV1{},
+			args: args{
+				ctx:    context.Background(),
+				data:   []byte("You cannot find the secret within"),
+				verify: true,
+			},
+			want:    nil,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := ScannerV1{}
+			got, err := s.FromData(tt.args.ctx, tt.args.verify, tt.args.data)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NpmTokenV1.FromData() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			for i := range got {
+				if len(got[i].Raw) == 0 {
+					t.Fatalf("no raw secret present: \n %+v", got[i])
+				}
+				got[i].Raw = nil
+			}
+			if diff := pretty.Compare(got, tt.want); diff != "" {
+				t.Errorf("NpmTokenV1.FromData() %s diff: (-got +want)\n%s", tt.name, diff)
+			}
+		})
+	}
+}
+
+func BenchmarkFromDataV1(benchmark *testing.B) {
```

Comment:
The main downside of consolidating everything under one package is that you can't have `BenchmarkFromData` multiple times.